lib/semian/lru_hash.rb

```diff
@@ -0,0 +1,61 @@
+class LRUHash
+  extend Forwardable
+  def_delegators :@table, :size, :count, :empty?, :keys, :clear, :delete
+  attr_reader :table, :minimum_time_in_lru
+
+  class NoopMutex
+    def synchronize(*)
```

Comment:
You're going to need to yield here, which leads me to believe you should probably make sure this code-path is hit :)